docs/HOL_Deploy_Private_Docker_Registry/README.md

```diff
@@ -0,0 +1,561 @@
+
+- You know how to set up Continuous Integration (CI) with Visual Studio Team Services (You don't? [Learn about CI]( https://microsoft.github.io/PartsUnlimitedMRP/fundvsts/fund-01-MS-CI.html))  
+
+**Tasks Overview**
+
+**1. Set up a private Docker registry** This will walk you through creating a private Docker registry. It also demonstrates how to tag, push and pull your images.   
+
+**2. Set up a Secured Continuous Integration (CI) with Visual Studio Team Services (VSTS)** Integrate Docker images and containers into your DevOps workflows using [Docker Integration](https://marketplace.visualstudio.com/items?itemName=ms-vscs-rm.docker) for Team Services. This Docker extension adds a task that enables you to build Docker images, push Docker images to an authenticated Docker registry, run Docker images or execute other operations offered by the Docker CLI.   
+
+###Task 1: Set up a private Docker registry
+**Step 1.** Deploy an Ubuntu VM with Docker Engine. [Do it quickly on Azure](https://azure.microsoft.com/en-us/documentation/templates/docker-simple-on-ubuntu/)
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fdocker-simple-on-ubuntu%2Fazuredeploy.json" target="_blank"> <img src="http://azuredeploy.net/deploybutton.png"/> </a>
+
+> **Note:** Latest Ubuntu OS version 16.04 LTS is recommended.
+
+**Step 2.** Once deployed, SSH into the Ubuntu VM using [PuTTY](http://www.chiark.greenend.org.uk/~sgtatham/putty/):
+
+![](<media/putty.png>)
+
+Enter your credentials to your VM:
+
+![](<media/entercredentials.png>)
+
+> **Note:** Run the following command to initiate the ssh connection if you are using a Unix machine: `ssh <user>@<public-ip>`
+
+**Step 3.** Install apache2-utils package. It contains the htpasswd utility which will be used to generate password hashes Nginx can understand:
+```
+$ sudo apt-get -y install apache2-utils
+```
+
+**Step 4.** Set up a Docker registry container. Let's create a new storage account to store Docker images, please sign in to the [Azure portal](https://ms.portal.azure.com/). On the Hub menu, select **New** -> **Storage** -> **Storage account**. Enter your details, and then click **Create**:
+
+![](<media/createstorageaccount.png>)
+
+ Once the storage account is created, click **Access Keys** setting, and then copy your storage account name and access key:    
+
+![](<media/storagenameandkey.png>)
+
+Add and save the following contents to docker-compose.yml file, and replace **storage-account** and **storage-key** with your storage account name and access key:
+```
+registry:
+  image: registry:2
+  ports:
+    - 127.0.0.1:5000:5000
+  environment:
+    - REGISTRY_STORAGE=azure
+    - REGISTRY_STORAGE_AZURE_ACCOUNTNAME="<storage-account>"
+    - REGISTRY_STORAGE_AZURE_ACCOUNTKEY="<storage-key>"
+    - REGISTRY_STORAGE_AZURE_CONTAINER="registry"
+  volumes:
+    - ./data:/data
+```
+
```

Comment:
Move the section (line 51 - 64) between the step 5. statement and before the mkdir command line (line 67)